frontend/packages/knative-plugin/src/components/services/ServiceDetailsPage.tsx

```diff
@@ -1,28 +1,106 @@
 import * as React from 'react';
+import * as _ from 'lodash';
+import { useTranslation } from 'react-i18next';
 import { useActivePerspective } from '@console/dynamic-plugin-sdk';
-import { DetailsForKind } from '@console/internal/components/default-resource';
+import { Conditions } from '@console/internal/components/conditions';
 import { DetailsPage } from '@console/internal/components/factory';
-import { navFactory } from '@console/internal/components/utils';
+import {
+  ContainerTable,
+  DetailsItem,
+  ExternalLink,
+  ResourceSummary,
+  SectionHeading,
+  navFactory,
+} from '@console/internal/components/utils';
+import { useK8sWatchResource } from '@console/internal/components/utils/k8s-watch-hook';
 import { K8sResourceKind, referenceForModel, K8sKind } from '@console/internal/module/k8s';
 import {
   ActionMenu,
   ActionMenuVariant,
   ActionServiceProvider,
   useTabbedTableBreadcrumbsFor,
 } from '@console/shared';
+import { RevisionModel } from '../../models';
 import { isServerlessFunction } from '../../topology/knative-topology-utils';
+import { RevisionKind, ServiceKind, ServiceTypeValue } from '../../types';
 import { serverlessTab } from '../../utils/serverless-tab-utils';
-import ServerlessFunctionType from '../overview/ServerlessFunctionType';
+import { KnativeServiceTypeContext } from '../functions/ServiceTypeContext';
+import RevisionsOverviewList from '../overview/RevisionsOverviewList';
+
+const ServiceDetails: React.FC<{ obj: ServiceKind }> = ({ obj }) => {
+  const { t } = useTranslation();
+  const [revisions, revisionLoaded, revisionErrorLoad] = useK8sWatchResource<RevisionKind[]>({
+    kind: referenceForModel(RevisionModel),
+    namespace: obj.metadata.namespace,
+    isList: true,
+  });
+
+  const revisionsForService =
+    revisionLoaded &&
+    !revisionErrorLoad &&
+    revisions.filter(
+      (revision) =>
+        _.get(revision.metadata, `labels["serving.knative.dev/service"]`) === obj.metadata.name,
+    );
```

Comment:
I think you can use a `labelSelector` in `useK8sWatchResource` to get only the matching revisions instead of loading all `Revisions` and then filter them on the client.